inference-engine/src/mkldnn_plugin/nodes/mkldnn_depth_to_space_node.cpp

```diff
@@ -0,0 +1,393 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "mkldnn_depth_to_space_node.h"
+#include <legacy/ie_layers.h>
+#include <string>
+#include <cmath>
+#include <mkldnn_types.h>
+#include <mkldnn_extension_utils.h>
+#include "ie_parallel.hpp"
+#include "common/cpu_memcpy.h"
+
+#define THROW_ERROR THROW_IE_EXCEPTION << "DepthToSpace layer with name '" << getName() << "' "
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+MKLDNNDepthToSpaceNode::MKLDNNDepthToSpaceNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
+        : MKLDNNNode(layer, eng, cache) {}
+
+void MKLDNNDepthToSpaceNode::getSupportedDescriptors() {
+    auto* depthToSpaceLayer = dynamic_cast<DepthToSpaceLayer*>(getCnnLayer().get());
+    if (depthToSpaceLayer == nullptr)
+        THROW_ERROR << "cannot convert from CNN layer";
+
+    SizeVector srcDims = depthToSpaceLayer->insData[0].lock()->getTensorDesc().getDims();
+    if (srcDims.size() < 3)
+        THROW_ERROR << "has incorrect number of input dimensions";
+    if (srcDims.size() > 5)
+        THROW_ERROR << "doesn't support dimensions with rank greater than 5";
+
+    SizeVector dstDims = depthToSpaceLayer->outData[0]->getTensorDesc().getDims();
+    if (srcDims.size() != dstDims.size())
+        THROW_ERROR << "has incorrect number of input/output dimensions";
+
+
+    std::string modeString = depthToSpaceLayer->GetParamAsString("mode");
+    if (modeString == "blocks_first") {
+        mode = DepthToSpaceMode::BLOCKS_FIRST;
+    } else if (modeString == "depth_first") {
+        mode = DepthToSpaceMode::DEPTH_FIRST;
+    } else {
+        THROW_ERROR << "doesn't support mode: " << modeString;
+    }
+
+    blockSize = depthToSpaceLayer->GetParamAsUInt("block_size", 1);
+    if (blockSize == 0)
+        THROW_ERROR << "Incorrect blockSize parameter is zero!";
+
+    size_t nSpatialDims = srcDims.size() - 2;
+    blockStep = static_cast<size_t>(std::pow(blockSize, nSpatialDims));
+    if (srcDims[1] % blockStep)
+        THROW_ERROR << "has block_size parameter which is incompatible with input tensor channels dimension size";
```

Comment:
I think `block_size` is more correct because the layer has got `block_size` parameter not `blockSize`. And there is `block_size` in [spec](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/movement/DepthToSpace_1.md)